kpops/component_handlers/schema_handler/schema_handler.py

```diff
@@ -44,13 +44,12 @@ def schema_provider(self) -> SchemaProvider:
     def load_schema_handler(
         cls, components_module: str | None, config: PipelineConfig
     ) -> SchemaHandler | None:
-        if not config.schema_registry_url:
-            return None
-
-        return cls(
-            url=config.schema_registry_url,
-            components_module=components_module,
-        )
+        if config.schema_registry.enabled:
+            return cls(
+                url=config.schema_registry.url,
```

Comment:
consider passing the entire `SchemaRegistryConfig`, we might extend it in the future